test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,356 @@
+// +build token_propagation_elasticsearch
+
+package e2e
+
+import (
+	"bytes"
+	goctx "context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/jaeger-client-go/config"
+	"golang.org/x/net/html"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis"
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	esv1 "github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1"
+)
+
+type TokenPropagationTestSuite struct {
+	suite.Suite
+}
+
+func (suite *TokenPropagationTestSuite) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
```

Comment:
Is there a reason this is openshift (es-operator) specific? Shouldn't it also work with any external ES cluster?

Reply:
The test is openshift specific, but it can work with an external ES cluster. but I assumed that the whole test environment is openshift, or at least the oauth part used with oauth proxy sidecar.

Reply:
The OAuth part is OpenShift-specific, but if we can make this test work without the OAuth Proxy, then we can test it in the Kubernetes test suite.